pkg/bundles/kontext/bundle_test.go

```diff
@@ -34,3 +40,94 @@ func TestBundleLayerIndex(t *testing.T) {
 		t.Errorf("Size() = %d, wanted %d", got, want)
 	}
 }
+
+func TestDockerIgnorableBundle(t *testing.T) {
+
+	wd, err := os.Getwd()
+	if err != nil {
+		t.Fatal("os.Getwd() =", err)
+	}
+
+	dockerIgnoreTests := map[string]struct {
+		dir  string
+		want []string
+		size int64
+	}{
+		"noStarExcludes": {
+			dir: filepath.Join(filepath.Dir(wd), "..", "ignore", "testdata", "dir2"),
+			want: []string{
+				"/var/run/kontext",
+				"/var/run/kontext/.dockerignore",
+				"/var/run/kontext/README.md",
+				"/var/run/kontext/pom.xml",
+				"/var/run/kontext/src",
+				"/var/run/kontext/src/main",
+				"/var/run/kontext/src/main/java",
+				"/var/run/kontext/src/main/java/One.java",
+				"/var/run/kontext/src/main/resources",
+				"/var/run/kontext/src/main/resources/application.properties",
+				"/var/run/kontext/target/foo-runner.jar",
+				"/var/run/kontext/tempABC",
+				"/var/run/kontext/tempABC/.gitkeep",
+			},
+			size: 536,
+		},
+		"starignore": {
+			dir: filepath.Join(filepath.Dir(wd), "..", "ignore", "testdata", "starignore"),
+			want: []string{
+				"/var/run/kontext",
+				"/var/run/kontext/README.md",
+				"/var/run/kontext/target/foo-runner.jar",
+				"/var/run/kontext/target/lib/one.jar",
+				"/var/run/kontext/target/quarkus-app/one.txt",
+			},
+			size: 251,
+		},
+	}
+
+	for name, tc := range dockerIgnoreTests {
+		t.Run(name, func(t *testing.T) {
+			//Bundle
+			l, err := bundle(tc.dir)
+			if err != nil {
+				t.Error("bundle() = ", err)
+			}
+
+			s, err := l.Size()
+			if err != nil {
+				t.Error("bundle#l.Size() = ", err)
+			}
+
+			if s != tc.size {
+				t.Errorf("Expected Size = %d, got %d", tc.size, s)
+			}
+
+			var actualBundleFiles []string
+
+			r, err := l.Uncompressed()
+			if err != nil {
+				t.Error("error reading layer image", err)
+			}
+			defer r.Close()
+
+			tr := tar.NewReader(r)
+			for {
+				h, err := tr.Next()
+				if errors.Is(err, io.EOF) {
+					break
+				}
+				if err != nil {
+					t.Error("error reading tar", err)
+					break
+				}
+
+				actualBundleFiles = append(actualBundleFiles, h.Name)
+			}
+
+			if !reflect.DeepEqual(tc.want, actualBundleFiles) {
+				t.Errorf("Expected %#v, got %#v", tc.want, actualBundleFiles)
```

Comment:
I actually thought of using this lib, but was not sure if the dep was there as part of the project , so left it over 😉

Reply:
@mattmoor  I think we need to trigger `update-codegen` to update the deps, as I see one failing test because of it https://github.com/mattmoor/mink/pull/308/checks?check_run_id=1556836949